pydatastructs/utils/_backend/cpp/GraphNode.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef GRAPH_NODE_HPP
+#define GRAPH_NODE_HPP
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include <string>
+
+typedef struct {
+    PyObject_HEAD
+    std::string name;
+    PyObject* data;
+} GraphNode;
```

Comment:
```suggestion
template <typename T>
typedef struct {
    PyObject_HEAD
    std::string name;
    T data;
} GraphNode<T>;
```

`float` (32, 64), `int` (8, 32, 64) , `string`, `char`, `complex`.